### PR TITLE
[BI-1055] Save Germplasm

### PR DESCRIPTION
**Pull request by ctucker3**
Depends on:
bi-web: BI-1055

Updates the germplasm import to save all of the required fields. 

There are some importer updates in there to better catch type errors and missing values. 

I added tests to try and test all error scenarios:
- Minimum import required fields, success
- Female parent not exist db id, throw error
- Female parent not exist entry number, throw error
- Male parent not exist db id, throw error
- Female parent not exist entry number, throw error
- Bad breeding method, throw error
- Some entry numbers, throw an error
- Numerical entry numbers
- Required fields missing, throw error
- Missing required headers
- Missing optional headers
- No entry numbers, automatic generation
- Full import, success
- Preview, non-preview fields not shown
- Male db, no female db id, pedigree string is null
- Circular parent dependency

### CODE REVIEW

**Review by HMS17 — CHANGES_REQUESTED** [review]

**Review by nickpalladino — APPROVED** [review]
I like all the tests. I did some happy path testing and data seems to be being written ok but there are a lot of requirements here so I'll have to leave the exhaustive checking to @DebWeigand.

File: src/main/resources/db/migration/V0.5.29__create_germplasm_sequence.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
```
[the remark this answers is not in the document]
Comment (ctucker3) [reply src/main/resources/db/migration/V0.5.29__create_germplasm_sequence.sql]: Good catch! I updated the germplasm sequence migration to be V0_5_30, leaving space for your migration at V0_5_29.

**Review by HMS17 — APPROVED** [review]

File: src/test/java/org/breedinginsight/brapps/importer/GermplasmTemplateMap.java [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+        assertEquals(testUser.getId().toString(), additionalInfo.getAsJsonObject("createdBy").get("userId").getAsString(), "Wrong createdBy userId");
+        // Created by User name
+        assertEquals(testUser.getName(), additionalInfo.getAsJsonObject("createdBy").get("userName").getAsString(), "Wrong createdBy userId");
+        // Breeding Method
+        String breedingMethodName = fileData.getString(i, "Breeding Method");
+        BreedingMethodEntity breedingMethod = breedingMethodDAO.findByNameOrAbbreviation(breedingMethodName).get(0);
+        assertEquals(breedingMethod.getId().toString(), additionalInfo.get("breedingMethodId").getAsString(), "Wrong Breeding Method ID");
+        assertEquals(breedingMethod.getName(), additionalInfo.get("breedingMethod").getAsString(), "Wrong Breeding Method name");
+        // Seed source
+        assertEquals(fileData.getString(i, "Source"), germplasm.get("seedSource").getAsString(), "Wrong seed source description");
```
Comment (HMS17) [inline src/test/java/org/breedinginsight/brapps/importer/GermplasmTemplateMap.java:597]: Should the failure message change now that we switched from seed source description to seed source?

Comment (ctucker3) [reply src/test/java/org/breedinginsight/brapps/importer/GermplasmTemplateMap.java]: updated

**Review by HMS17 — APPROVED** [review]

**Pull request merged**